### ISSUE TITLE
implement docker_pull rule

### ISSUE THREAD

**Issue by mikedanese** [bazelbuild/bazel#1058]
I've seen hints that this is planned as future work but I'm not aware of an issue tracking it. Is anyone actively working on this? Is there anything blocking this?

**Comment by damienmg** [comment]
Yes that is in plan (last step of http://goo.gl/OZV3o0).

It will comes after #699 which should come soon.

**Comment by damienmg** [comment]
I have a prototype that's a few lines long but it relies on the docker daemon:
```
  repository_ctx.file("BUILD", """
load("@bazel_tools//tools/build_defs/docker:docker.bzl", "docker_build")

docker_build(
    name = "image",
    base = ":base.tar",
    visibility = ["//visibility:public"],
)
""")
```

In the full version I would like to do it outside of docker.

**Comment by yugui** [comment]
* Why do we need to wait for #699? Does it mean you are trying to implement the repository rule with [docker-py](https://pypi.python.org/pypi/docker-py)?
* How can I help you to make it happen?

**Comment by damienmg** [comment]
@yugui it does not needs to wait, it's just less a priority than #699.

To make it happens, I am waiting for some code to get OSS then this can be written without requiring docker.

**Comment by yugui** [comment]
Thank you for your reply.  I understand.
I'm happy to help if there's something I can do for the rule.

**Comment by mattmoor** [comment]
Please give this a try: https://github.com/bazelbuild/rules_docker#docker_pull
